File: src/mmu/no_mbc.c
#include <stdint.h>

#include "gbemu/mmu.h"

uint8_t no_mbc_read_rom(uint16_t loc) { return rom[loc]; }

uint8_t no_mbc_read_ram(uint16_t loc __attribute__((unused))) { return 0xFF; }

void no_mbc_write_rom(uint16_t loc __attribute__((unused)),
                      uint8_t val __attribute__((unused))) {}

void no_mbc_write_ram(uint16_t loc __attribute__((unused)),
                      uint8_t val __attribute__((unused))) {}

File: rust/src/mmu/no_mbc.rs
use super::Mmu;

impl Mmu {
    pub fn no_mbc_read_rom(&self, loc: u16) -> u8 {
        self.rom[loc as usize]
    }
    pub fn no_mbc_read_ram(&self, _loc: u16) -> u8 {
        0xFF
    }
    pub fn no_mbc_write_rom(&self, _loc: u16, _val: u8) {}
    pub fn no_mbc_write_ram(&self, _loc: u16, _val: u8) {}
}
